src/app/user/modules/records/components/record-form/record-form.component.ts

```diff
@@ -71,6 +79,42 @@ export class RecordFormComponent implements OnInit {
       this.numHours.setValue(offCampusEvent.num_hours);
       this.numParticipants.setValue(offCampusEvent.num_participants);
     });
+    if (this.route.snapshot.queryParams.record_id !== undefined) {
+      const recordID = this.route.snapshot.queryParams.record_id;
+      this.recordService.getRecordWithDetail(recordID).subscribe(
+        record => {
+          this.toUpdate = true;
+          this.record = record;
+          this.record.off_campus_event = record.off_campus_event as OffCampusEvent;
```

Comment:
I haven't try it yet, but I think it might be possible to set value with an object, then we don't have to repeat this process many times.